### PR TITLE
Utils and Matutils changes

### PR DESCRIPTION
**Pull request by bhargavvader**
1. Moved `dirichlet_expectation` to `matutils` as the same code was being used in both `ldamodel` and `hdpmodel`.
2. Changed all `numpy` in `utils` and `matutils` to `np`, keeping it uniform across the package.
3. Fixed hanging indent.
4. Changed import conventions in `hdpmodel` to make it consistent with `ldamodel`

### CODE REVIEW

File: gensim/utils.py [hunk 1]
```diff
@@ -608,7 +607,7 @@ def is_corpus(obj):
             doc1 = next(iter(obj))  # empty corpus is resolved to False here
         if len(doc1) == 0:  # sparse documents must have a __len__ function (list, tuple...)
             return True, obj  # the first document is empty=>assume this is a corpus
-        id1, val1 = next(iter(doc1))  # if obj is a numpy array, it resolves to False here
+        id1, val1 = next(iter(doc1))  # if obj is a np array, it resolves to False here
```
Comment (piskvorky) [inline gensim/utils.py]: This replacement (and the one above) look unnecessary, the original was clearer.

Comment (bhargavvader) [inline gensim/utils.py]: Cool, will fix it

File: gensim/models/hdpmodel.py [hunk 2]
```diff
@@ -74,15 +64,15 @@ def expect_log_sticks(sticks):
 
 def lda_e_step(doc_word_ids, doc_word_counts, alpha, beta, max_iter=100):
     gamma = np.ones(len(alpha))
-    expElogtheta = np.exp(dirichlet_expectation(gamma))
+    expElogtheta = np.exp(matutils.dirichlet_expectation(gamma))
```
Comment (tmylk) [inline gensim/models/hdpmodel.py]: better to add `from matutils import dirichlet_expectation`

Comment (bhargavvader) [inline gensim/models/hdpmodel.py]: Okay, changed

**Comment by tmylk** [comment]
Thanks for the improvement!

File: gensim/utils.py [hunk 3]
```diff
@@ -829,15 +828,15 @@ def __init__(self, q, corpus, chunksize, maxsize, as_numpy):
 
     def run(self):
         if self.as_numpy:
-            import numpy # don't clutter the global namespace with a dependency on numpy
+            import np # don't clutter the global namespace with a dependency on numpy
```
Comment (piskvorky) [inline gensim/utils.py]: Looks like the global module namespace already has numpy, so this line is unnecessary.

File: gensim/utils.py [hunk 4]
```diff
@@ -801,13 +800,13 @@ def chunkize_serial(iterable, chunksize, as_numpy=False):
     [[0, 1, 2], [3, 4, 5], [6, 7, 8], [9]]
 
     """
-    import numpy
+    import numpy as np
```
Comment (piskvorky) [inline gensim/utils.py]: Already imported at module level, this line no longer unnecessary.

**Pull request merged**